Exercise1-1

asv-tungpt opened the pull request.

asv-tungpt committed:
Exercise1-1

asv-tungpt committed:
[Update 1] Exercise 1-1

asv-tungpt committed:
[Update 2] Exercise 1-1

vn-trinhtk commented on djangoguestbook/djangoguestbook/settings.py:
```diff
@@ -0,0 +1,164 @@
+# Django settings for djangoguestbook project.
```
Không commit những file *.pyc

vn-trinhtk commented on djangoguestbook/guestbook/models.py:
```diff
@@ -0,0 +1,18 @@
+__author__ = 'fatatoopc'
```
bỏ hết **author** = 'fatatoopc'

vn-trinhtk commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,45 @@
+
+def sign_post(request):
+    if request.method == 'POST':
+        logging.warning("access!!!!!!!!!@#!@$#%^$&^*%$%^&*^%$#%^&%^$")
```
Log xài xong thì xoá → remove thư viện logging luôn.

phucnpm commented on djangoguestbook/hello/templates/index.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
lưu ý: 
- không commit những file thừa không thuộc dự án (bài tập)

phucnpm commented on djangoguestbook/hello/urls.py:
```diff
@@ -0,0 +1,6 @@
+__author__ = 'fatatoopc'
+from django.conf.urls import *
```
file thừa

asv-tungpt committed:
[Update 3] Exercise 1-1

phucnpm commented on djangoguestbook/hello/views.py:
```diff
@@ -0,0 +1,8 @@
+__author__ = 'fatatoopc'
+from django.shortcuts import render_to_response, get_object_or_404
```
file thừa

phucnpm commented on djangoguestbook/djangoguestbook/wsgi.py:
```diff
@@ -0,0 +1,32 @@
+"""
+WSGI config for djangoguestbook project.
```
file này có cần thiết ?

phucnpm commented on djangoguestbook/djangoguestbook/urls.py:
```diff
@@ -0,0 +1,20 @@
+from django.conf.urls import patterns, include, url
+
```
thử move 
-  /djangoguestbook/djangoguestbook/urls.py
-  /djangoguestbook/djangoguestbook/settings.py

ra ngoài 
-  /djangoguestbook

phucnpm commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,41 @@
+    # a slight chance that Greeting that had just been written would not
+    # show up in a query.
+    greetings_query = Greeting.query(ancestor=guestbook_key(guestbook_name)).order(-Greeting.date)
+    greetings = greetings_query.fetch(10)
```
hãy dùng appengine.api memcache

asv-tungpt committed:
[Update 4] Exercise 1-1

asv-tungpt committed:
[Update 5] Exercise 1-1

asv-tungpt committed:
[Update 6] Exericise 1-1

Xoá những file và dòng lệnh

vn-trinhtk commented on djangoguestbook/guestbook/models.py:
```diff
@@ -0,0 +1,18 @@
+# entity group. Queries across the single entity group will be consistent.
+# However, the write rate should be limited to ~1/second.
+
+def guestbook_key(guestbook_name=DEFAULT_GUESTBOOK_NAME):
```
Chỗ này mình nghĩ nên đổi tên hàm cho dễ hiểu một chút, get_guestbook_key chẳng hạn

vn-trinhtk commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,55 @@
+
+def main_page(request):
+    guestbook_name = request.GET.get('guestbook_name', DEFAULT_GUESTBOOK_NAME)
+    greetings_query = Greeting.query(ancestor=guestbook_key(guestbook_name)).order(-Greeting.date)
```
Cái này còn sử dụng không ?

phucnpm commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,55 @@
+            logging.error('Memcache set failed.')
+    return greetings
+
+def render_greetings(guestbook_name):
```
render là thao tác hiển thị lên UI, không đặt tên hàm gây nhầm lẫn

phucnpm commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,55 @@
+        return greetings
+    else:
+        greetings = render_greetings(guestbook_name)
+        if not memcache.add('%s:greetings' % guestbook_name, greetings, 10):
```
- dùng memcard.set thay cho add, tránh việc conflict Id khi nhiều user thao tác cùng lúc
- hãy suy nghĩ thêm thời điểm tạo / update memcache

asv-tungpt committed:
[Update 7] Exercise 1-1

Chỉnh sửa lại memcache

asv-tungpt committed:
[Update 8] Exercise 1-1

Fix memcache

phucnpm commented:
remove tất cả file *.pyc khỏi git 
`git rm`

phucnpm commented:
xoá thư mục djangoguestbook/djangoguestbook vì không cần thiết

phucnpm commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,53 @@
+from guestbook.models import Greeting, get_guestbook_key, DEFAULT_GUESTBOOK_NAME
+import urllib
+import logging
+from google.appengine.api import memcache
```
sắp xêp thứ tự import thư viện theo thứ tự
- Standard module
- 3rd party modules (appengine, django, pytest, such as mock)
- Application module

phucnpm commented on djangoguestbook/guestbook/views.py:
```diff
@@ -0,0 +1,53 @@
+    else:
+        greetings_query = Greeting.query(ancestor=get_guestbook_key(guestbook_name)).order(-Greeting.date)
+        greetings = greetings_query.fetch(10)
+        if not memcache.add('%s:greetings' % guestbook_name, greetings, 10):
```
dùng `memcache.set` thay cho `add` method

phucnpm commented on djangoguestbook/guestbook/__init__.py:
```diff
@@ -0,0 +1 @@
+__author__ = 'fatatoopc'
```
- xoá author
- tắt chế độ tự động chèn author trong Pycharm

asv-tungpt committed:
[Update 9] Exercise 1-1

-Xoá các file *.pyc và các thư mục, file không cần thiết
-Sử dụng memcache.set thay vì memcache.add
-Sắp xếp lại các import thư viện
-Tắt chèn tự động author trong pycharm

phucnpm commented:
vẫn còn file .pyc nhé 
![screen shot 2015-01-22 at 5 25 32 pm](https://cloud.githubusercontent.com/assets/7623464/5852537/db00dcf4-a25b-11e4-9a40-d0161417306c.png)

asv-tungpt committed:
[Update 10] Exercise 1-1

Xoá các file *.pyc:
    __init__.pyc
    models.pyc
    urls.pyc
    views.pyc
    main.pyc
